src/main/java/net/fabricmc/fabric/api/events/network/ConnectionEvent.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.events.network;
+
+import net.fabricmc.fabric.util.HandlerArray;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.TextComponent;
+
+public class ConnectionEvent {
```

Comment:
Can we split this into ClientConnectionEvent and ServerConnectionEvent, with clearer event names CONNECTED and DISCONNECTED?

Reply:
My worry about this, and the reason I went for the longer names, is that I'm not actually sure how much clearer the shorter names are. I think there's a potential for ambiguity about whether `ServerConnectionEvent` is fired on the server or clients (when connecting to a server).

I guess one has the same issue with packets (does client refer to the sender or receiver), so we have a pattern already.